orchestrator/careplanservice/handle_getpatient.go

```diff
@@ -0,0 +1,185 @@
+package careplanservice
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	fhirclient "github.com/SanteonNL/go-fhir-client"
+	"github.com/SanteonNL/orca/orchestrator/lib/auth"
+	"github.com/SanteonNL/orca/orchestrator/lib/coolfhir"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+	"net/http"
+	"net/url"
+)
+
+// handleGetPatient fetches the requested Patient and validates if the requester has access to the resource (is a participant of one of the CareTeams associated with the patient)
+// if the requester is valid, return the Patient, else return an error
+// Pass in a pointer to a fhirclient.Headers object to get the headers from the fhir client request
+func (s *Service) handleGetPatient(ctx context.Context, id string, headers *fhirclient.Headers) (*fhir.Patient, error) {
+	// fetch Patient + CareTeam, validate requester is participant of CareTeam
+	var patient fhir.Patient
+	err := s.fhirClient.Read("Patient/"+id, &patient, fhirclient.ResponseHeaders(headers))
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the CarePlan for which the Patient is the subject, get the CareTeams associated with the CarePlan
+	// The search for CarePlan already checks auth for CareTeam, so if we get back a valid CarePlan, we can assume the user has access to the patient
+	// If the user is not part of any of these CareTeams, the search for CarePlan will return an error
+	bundle, err := s.handleSearchCarePlan(ctx, url.Values{"subject-identifier": []string{patientBSN(&patient)}}, headers)
```

Comment:
@JorisHeadease why can't we use the `patient` search parameter of `CarePlan` again?

```

patient | reference | Who the care plan is for | CarePlan.subject.where(resolve() is Patient)(Patient)
-- | -- | -- | --

patient	[reference](https://www.hl7.org/fhir/R4/search.html#reference)	Who the care plan is for	CarePlan.subject.where(resolve() is Patient)
([Patient](https://www.hl7.org/fhir/R4/patient.html))
```